src/wren_core.c

```diff
@@ -659,10 +659,10 @@ DEF_NATIVE(list_iterate)
 
   if (!validateInt(vm, args, 1, "Iterator")) return PRIM_ERROR;
 
-  int index = (int)AS_NUM(args[1]);
+  uint32_t index = (uint32_t)AS_NUM(args[1]);
 
   // Stop if we're out of bounds.
-  if (index < 0 || index >= list->count - 1) RETURN_FALSE;
+  if (index >= list->count - 1) RETURN_FALSE;
```

Comment:
For the sake of clearness, I agree with you.

However, casting a negative value to unsigned should give a value greater than the list length, so we can get rid of a comparison (with a minor performance improvement).

However, I'm in favour of maintaining clearness over (minor) optimizations.

Should I revert this?